Add handling of list write to remove mutation

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#62904 Add handling of list write to remove mutation**
* #62903 Add tensor-scalar op

Differential Revision: [D30168493](https://our.internmc.facebook.com/intern/diff/D30168493)

Commit:
Add handling of list write to remove mutation

[ghstack-poisoned]

Krovatkin commented on torch/csrc/jit/passes/remove_mutation.cpp:
```diff
@@ -68,11 +69,27 @@ Node* MutationRemover::createSpecialMappedOp(Node* n) {
+    return false;
+  }
+  int64_t index = *constant_as<int64_t>(n->input(1));
+  if (index < 0) {
```
nitpick: do we have `normalizeIndex` somewhere?

eellison replied:
let me check..

Krovatkin reviewed (approved):
:shipit:

Commit:
Update on "Add handling of list write to remove mutation"




[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30168493).

Commit:
Update on "Add handling of list write to remove mutation"


Differential Revision: [D30168493](https://our.internmc.facebook.com/intern/diff/D30168493)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30168493).

Commit:
Update on "Add handling of list write to remove mutation"


Differential Revision: [D30168493](https://our.internmc.facebook.com/intern/diff/D30168493)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30168493).

Commit:
Update on "Add handling of list write to remove mutation"


Differential Revision: [D30168493](https://our.internmc.facebook.com/intern/diff/D30168493)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30168493).
